src/Meditation.Core/Services/AttachableProcessListProvider.cs

```diff
@@ -1,24 +1,89 @@
-﻿using Meditation.Common.Models;
+﻿using CliWrap;
+using Meditation.Common.Models;
 using Meditation.Common.Services;
 using Microsoft.Diagnostics.NETCore.Client;
+using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
 
 namespace Meditation.Core.Services
 {
     internal class AttachableProcessListProvider : IAttachableProcessListProvider
     {
-        private readonly ImmutableArray<ProcessInfo> attachedProcesses;
+        private readonly IProcessListProvider processListProvider;
+        private Task<ImmutableArray<ProcessInfo>> attachableProcessesTask;
 
         public AttachableProcessListProvider(IProcessListProvider processListProvider)
+        {
+            this.processListProvider = processListProvider;
+            attachableProcessesTask = LoadAttachableProcessesAsync();
+        }
+
+        public Task<ImmutableArray<ProcessInfo>> GetAllAttachableProcessesAsync()
+            => attachableProcessesTask;
+
+        public void Refresh()
+        {
+            processListProvider.Refresh();
+            attachableProcessesTask = LoadAttachableProcessesAsync();
+        }
+
+        private async Task<ImmutableArray<ProcessInfo>> LoadAttachableProcessesAsync()
+        {
+            var builder = new List<ProcessInfo>();
+            var netCoreProcesses = LoadNetCoreProcesses();
+            var netCoreProcessesLookup = netCoreProcesses.ToDictionary(p => p.Id, p => p);
+
+            foreach (var netFrameworkProcess in await LoadNetFrameworkProcessesAsync())
+            {
+                // NET Framework process obtaining logic might not be 100% reliable, trust .NET Core client primarily
+                if (netCoreProcessesLookup.ContainsKey(netFrameworkProcess.Id))
+                    continue;
+
+                builder.Add(netFrameworkProcess);
+            }
+
+            builder.AddRange(netCoreProcessesLookup.Values);
+            return builder.ToImmutableArray();
+        }
+
+        private async Task<ImmutableArray<ProcessInfo>> LoadNetFrameworkProcessesAsync()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                ImmutableArray.Create<ProcessInfo>();
+
+            var builder = new List<ProcessInfo>();
+            var stdout = new StringBuilder();
+            await Cli.Wrap("cmd")
+                .WithArguments("/c \"tasklist /m \"mscorlib*\" /fo csv /nh\"")
```

Comment:
I suggest to put such strings "/c \"tasklist /m \"mscorlib*\" /fo csv /nh\"" into static fields or separate class with only fields like that